src/Core/Startup/ErrorDirector.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace SilverStripe\Core\Startup;
+
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\Core\Kernel;
+use SilverStripe\Security\Permission;
+use SilverStripe\Security\Security;
+
+/**
+ * Specialised Director class used by ErrorControlChain to handle error and redirect conditions
```

Comment:
Can we mark this whole class with `@internal This is an internal support class and may be changed/removed in minor releases`? I wouldn't want to be supporting people's subclasses of this.